tests/testthat/test-vivax-compartmental.R

```diff
@@ -0,0 +1,161 @@
+test_that('vivax ODE stays at equilibrium with a constant total_M', {
+  parameters <- get_parameters(list(
+    individual_mosquitoes = TRUE
+  ), parasite = "vivax")
+  total_M <- 1000
+  timesteps <- 365 * 10
+  f <- parameters$blood_meal_rates
+  models <- parameterise_mosquito_models(parameters, timesteps)
+  solvers <- parameterise_solvers(models, parameters)
+  
+  counts <- c()
+  
+  for (t in seq(timesteps)) {
+    counts <- rbind(counts, c(t, solvers[[1]]$get_states()))
+    aquatic_mosquito_model_update(
+      models[[1]]$.model,
+      total_M,
+      f,
+      parameters$mum
+    )
+    solvers[[1]]$step()
+  }
+  
+  expected <- c()
+  equilibrium <- initial_mosquito_counts(
+    parameters,
+    1,
+    parameters$init_foim,
+    parameters$total_M
+  )[ODE_INDICES]
+  for (t in seq(timesteps)) {
+    expected <- rbind(expected, c(t, equilibrium))
+  }
+  
+  expect_equal(counts, expected, tolerance=1e-4)
+})
+
+test_that('vivax Adult ODE stays at equilibrium with a constant foim and mu', {
+  parameters <- get_parameters(parasite = "vivax")
+  parameters <- set_equilibrium(parameters, 100.)
+  f <- parameters$blood_meal_rates
+  timesteps <- 365 * 10
+  models <- parameterise_mosquito_models(parameters, timesteps)
+  solvers <- parameterise_solvers(models, parameters)
+  
+  counts <- c()
+  
+  for (t in seq(timesteps)) {
+    states <- solvers[[1]]$get_states()
+    counts <- rbind(counts, c(t, states))
+    adult_mosquito_model_update(
+      models[[1]]$.model,
+      parameters$mum,
+      parameters$init_foim,
+      states[ADULT_ODE_INDICES['Sm']],
+      f
+    )
+    solvers[[1]]$step()
+  }
+  
+  expected <- c()
+  equilibrium <- initial_mosquito_counts(
+    parameters,
+    1,
+    parameters$init_foim,
+    parameters$total_M
+  )
+  
+  for (t in seq(timesteps)) {
+    expected <- rbind(expected, c(t, equilibrium))
+  }
+  expect_equal(counts, expected, tolerance=1e-3)
+})
+
+test_that('vivax ODE stays at equilibrium with low total_M', {
```

Comment:
This is perhaps redundant? Is there any change in the vector model for vivax?

Reply:
I think there was a while when I wasn't sure, but I think you're right: I can take them out.